packages/node/src/nodeClient.ts

```diff
@@ -9,29 +9,52 @@ export class NodeClient implements Client<Options> {
   /** Options for the client. */
   protected readonly _options: Options;
 
+  private _events: Event[];
+  private _transport: Transport;
+
   /**
    * Initializes this client instance.
    *
    * @param apiKey API key for your project
    * @param options options for the client
    */
-  public constructor(apiKey: string, options: Options) {
+  public constructor(apiKey: string, options: Options = {}) {
     this._apiKey = apiKey;
     this._options = options;
+    this._events = [];
+    this._transport = this._setupTransport();
+
+    const uploadIntervalInSec = options.uploadIntervalInSec ?? 30;
+    var self = this;
+    setInterval(function() {
+      self.flush();
+    }, uploadIntervalInSec * 1000);
   }
 
   /**
    * @inheritDoc
    */
-  getOptions(): Options {
+  public getOptions(): Options {
     return this._options;
   }
 
   /**
    * @inheritDoc
    */
-  flush(): void {
-    throw new Error('Method not implemented.');
+  public flush(): void {
+    // Check if there's 0 events, flush is not needed.
+    const arraryLength = this._events.length;
+    if (arraryLength === 0) {
+      return;
+    }
+
+    const response = this._transport.sendPayload(this._getCurrentPayload());
+    response.then(res => {
```

Comment:
Can we use `async`/`await` instead of promise chaining?

Reply:
async / await for?

Reply:
Are you asking when async/await is preferred? Or how and where to use it?

Reply:
How to use it? :)

Reply:
That's a big topic.

I would start here: https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await